074-connector-offsets-support.md

```diff
@@ -0,0 +1,180 @@
+    sink:
+      - partition:
+          kafkaTopic: "my-topic"
+          kafkaPartition: 1
+        offset:
+          kafkaOffset: 0
+```
+
+Notes:
+* Only one of the `source` and `sink` properties will be present.
+* The operator will update the `lastUpdateTime` whenever it updates the offsets section of the status.
+* If the `showOffsets` property is set to `false` the operator will leave the last observed offsets in the status, keeping the `lastUpdateTime` so the user sees the offsets list has not been updated.
+
+### Altering offsets
+To alter offsets a new `KafkaConnectorOffsets` custom resource will be added.
+This is similar to the approach used for Cruise Control with `KafkaRebalance`.
+This approach is used because the request requires a complex object to specify the desired offsets.
+
+The new resource should look like:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnectorOffsets
+metadata:
+  name: my-source-connector-offsets
+  labels:
+    # The strimzi.io/cluster label identifies the KafkaConnect instance
+    # in which this connector is running
+    strimzi.io/cluster: my-connect-cluster
+spec:
+  connectorName: my-source-connector
+  sourceOffsets:
+    - partition: "{\"connector-determined-partition\": \"1\"}" # value is a String containing JSON can be directly passed to the API
+      offset: "{\"connector-determined-offset\": \"1\"}" # value is a String containing JSON can be directly passed to the API
+  sinkOffsets:
+    - partition:
+        kafkaTopic: "my-topic"
+        kafkaPartition: 1
+      offset:
+        kafkaOffset: 0
+```
+
+Notes:
+* Only one of `sourceOffsets` and `sinkOffsets` should be specified.
+* The format matches the format shown in the status when listing offsets so it can be copy pasted here.
+* Both the `offset` properties are optional since the API allows these to be passed as `null` (this resets the offsets for that partition).
+* When the offset patch has been completed the status will have a `Ready` condition of `true`.
+* If the request fails the status will include the response from the API.
+* Strimzi will not attempt to verify that the connector is already stopped, it will rely on the API failing if that is the case.
+  This is to simplify the logic and means we don't have to account for what happens if the KafkaConnector spec has been updated but the connector isn't stopped yet.
+
+### Resetting offsets
+
+To reset the offsets for a particular connector the user will annotate the KafkaConnector resource with a new annotation: `strimzi.io/offsets-reset=true`.
+When the reset has been completed the operator will remove this annotation from the resource.
+
+If the reset fails the operator will add a condition to the KafkaConnector resource called `strimzi.io/offsets-reset` to communicate the failure and remove the annotation.
+
+Strimzi will shortcut and automatically fail to do the reset if the KafkaConnector resource is not stopped.
```

Comment:
Could Strimzi stop the connector if needed?

Reply:
It could, however the reason I chose not to do that is because I don't think it's best practice for the Strimzi operator to update the spec in the KafkaConnector resource. And it would need to change the `state` to `stopped`